azure/scope/machine_test.go

```diff
@@ -313,30 +316,51 @@ func TestMachineScope_InboundNatSpecs(t *testing.T) {
 					},
 				},
 				ClusterScoper: &ClusterScope{
+					AzureClients: AzureClients{
+						EnvironmentSettings: auth.EnvironmentSettings{
+							Values: map[string]string{
+								auth.SubscriptionID: "123",
+							},
+						},
+					},
 					AzureCluster: &infrav1.AzureCluster{
 						Spec: infrav1.AzureClusterSpec{
+							ResourceGroup:  "my-rg",
+							SubscriptionID: "123",
 							NetworkSpec: infrav1.NetworkSpec{
 								APIServerLB: infrav1.LoadBalancerSpec{
 									Name: "foo-loadbalancer",
+									FrontendIPs: []infrav1.FrontendIP{
+										{
+											Name: "foo-frontend-ip",
+										},
+									},
 								},
 							},
 						},
 					},
 				},
 			},
-			want: []azure.InboundNatSpec{
-				{
-					Name:             "machine-name",
-					LoadBalancerName: "foo-loadbalancer",
+			want: []azure.ResourceSpecGetter{
+				&inboundnatrules.InboundNatSpec{
+					Name:                      "machine-name",
+					LoadBalancerName:          "foo-loadbalancer",
+					ResourceGroup:             "my-rg",
+					FrontendIPConfigurationID: to.StringPtr(azure.FrontendIPConfigID("123", "my-rg", "foo-loadbalancer", "foo-frontend-ip")),
+					ExistingRules:             []converters.ExistingInboundNatSpec{},
 				},
 			},
 		},
 	}
 	for _, tt := range tests {
+		tt := tt
 		t.Run(tt.name, func(t *testing.T) {
-			if got := tt.machineScope.InboundNatSpecs(); !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("InboundNatSpecs() = %v, want %v", got, tt.want)
-			}
+			g := NewWithT(t)
+
+			t.Parallel()
+			result := tt.machineScope.InboundNatSpecs([]converters.ExistingInboundNatSpec{})
+			fmt.Printf("InboundNatSpecs() = %+v, want %+v\n", result, tt.want)
```

Comment:
Do we want to print the expectations always? Maybe this `fmt.Printf` should be a second argument to the `BeEquivalentTo()` on the next line, so it only prints when a test case fails. WDYT?

Reply:
Oh that was a debugging statement, I'll go ahead and just remove that.

Reply:
Actually I'm not sure how helpful that line is. Since it's returning a ResourceSpecGetter, the output just looks something like this. Do you have any suggestions on what to do here?
``` 
InboundNatSpecs([]converters.ExistingInboundNatSpec{}) = [0xc000112050], want [0xc0004812c0]
```